torch/distributed/_shard/sharding_plan/api.py

```diff
@@ -0,0 +1,25 @@
+import abc
+import torch.nn as nn
+
+from dataclasses import dataclass
+from typing import Dict
+
+from torch.distributed._shard.sharding_spec import ShardingSpec
+
+@dataclass
+class ShardingPlan(object):
+    """
+    Representation of a sharding plan.
+    Dict keyed by FQN of parameter to PlacementSpec.
+    """
+    plan: Dict[str, ShardingSpec]
+
+
+class ShardingPlanner(abc.ABC):
+    """
+    Default ShardingPlanner interface, can be extended and
+    implement advanced sharding strategies.
+    """
+    @abc.abstractmethod
```

Comment:
Is it possible that we give users an option to not shard some part of the module?

Reply:
yeah, user could manually pass a ShardingPlan that only shards some certain parameters.